### ISSUE TITLE
RFID Block Code Fix

### ISSUE THREAD

**Issue by CarsonGracey** [parallaxinc/BlocklyProp#290]
The comments have the correct code for each RFID block and their purpose.

```C
#include "simpletools.h"
#include "rfidser.h"      
                       
rfidser *rfid;                          //code inserted by "RFID open" block under include files

int main() 
{
  rfid = rfid_open(1, 2);               //code inserted by "RFID enable" block which has two arguments (RFID /enable pin, RFID SOUT pin)
  char *str = rfid_get(rfid, 1000);     //code inserted by "RFID read" block
  print("ID = %s.\n", str);             //no block - this is user code that already works

rfid_disable(rfid);                     //code inserted by "RFID disable" stops tag scanning, but keeps cog running 
rfid_close(rfid);                       //code inserted by "RFID close" block which disables and clears the cog                      
}
```

**Comment by DarkmatterVale** [comment]
Do you want me to add a "store in the variable ___" option in the RFID read block, or do you want me to leave it as a value?

**Comment by kgracey** [comment]
I’m not sure I understand the difference. 

Variable or value? Can you explain a bit more?

Ken Gracey
graceyken@gmail.com

**Comment by DarkmatterVale** [comment]
A variable will be simpler for the user, but more restrictive (only variables can be used). Although, I think that there will never be a need for values...

I will prototype the variables and send it to demo where you can take a look and make a better educated guess.

**Comment by kgracey** [comment]
Aha, I got ya now. A variable vs. a constant. Indeed, just a variable block will take care of this one. 

How are you going to handle 10 bytes of data?

Ken Gracey
graceyken@gmail.com

**Comment by DarkmatterVale** [comment]
What do you mean? Isn't there no need to handle 10 bytes of data for the RFID read block?

**Comment by kgracey** [comment]
The RFID Reader sends out 10 bytes of data. How could we handle all the data coming from the RFID Reader in Blockly?

**Comment by DarkmatterVale** [comment]
Can't we just store the data in a char as shown in your example?

```
int main() 
{
  rfid = rfid_open(1, 2);               //code inserted by "RFID enable" block which has two arguments (RFID /enable pin, RFID SOUT pin)
  char *str = rfid_get(rfid, 1000);     //code inserted by "RFID read" block
  print("ID = %s.\n", str);
```

**Comment by DarkmatterVale** [comment]
Or do I need to do something else to be able to handle the data?

**Comment by kgracey** [comment]
Sure, I guess so! I don’t know enough about C, but it seems that it auto-sizes so that if we get a whole bunch of bytes from the RFID reader it’ll just be one long Char, right?

**Comment by DarkmatterVale** [comment]
To be honest with you I have no idea. Its been a few months since I programmed C++, and at least a half a year since I last programmed C...

**Comment by kgracey** [comment]
Then let’s try this out. As long as we can use the output of the RFID reader to make some decisions in Blockly it’ll work fine. Let’s see what happens.

**Comment by DarkmatterVale** [comment]
Sounds like a great plan

**Comment by PropGit** [comment]
Is this issue verified fixed?

**Comment by CarsonGracey** [comment]
Tested and Verified. (the RFID close fix was here #338 )
